plugins/de.cognicrypt.codegenerator.tests/src/de/cognicrypt/codegenerator/generator/test/DefaultTasksGeneratorTest.java

```diff
@@ -0,0 +1,140 @@
+	CodeGenerator generatorHybridEnc;
+	CodeGenerator generatorDigitalSIgn;
+	Task encTask;
+	Task secPasswordTask;
+	Task LTATask;
+	Task secMPCompTask;
+	Task hybridEncTask;
+	Task digitalSignTask;
+	Configuration configEnc;
+	Configuration configSecPassword;
+	Configuration configLTA;
+	Configuration configSecMPComp;
+	Configuration configHybridEnc;
+	Configuration configDigitalSign;
+	DeveloperProject developerProject;
+	static int counter = 0;
+
+	@After
+	public void tearDown() throws CoreException {
+		TestUtils.deleteProject(testJavaProject.getProject());
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		DefaultTasksGeneratorTest.counter++;
+		this.testJavaProject = TestUtils.createJavaProject("TestProject_"+counter);
+		TestUtils.generateJavaClassInJavaProject(testJavaProject, "testPackage", "Test");
+		
+		this.encTask = TestUtils.getTask("SymmetricEncryption");
+		this.generatorEnc = new XSLBasedGenerator(testJavaProject.getProject(), encTask.getXslFile());
+		
+		this.secPasswordTask = TestUtils.getTask("SecurePassword");
+		this.generatorSecPassword = new XSLBasedGenerator(testJavaProject.getProject(), secPasswordTask.getXslFile());
+		
+		this.LTATask = TestUtils.getTask("LongTermArchiving");
+		this.generatorLTA = new XSLBasedGenerator(testJavaProject.getProject(), LTATask.getXslFile());
+		
+		this.secMPCompTask = TestUtils.getTask("SECMUPACOMP");
+		this.generatorSecMPComp = new XSLBasedGenerator(testJavaProject.getProject(), secMPCompTask.getXslFile());
+		
+		this.hybridEncTask = TestUtils.getTask("HybridEncryption");
+		this.generatorHybridEnc = new XSLBasedGenerator(testJavaProject.getProject(), hybridEncTask.getXslFile());
+		
+		this.digitalSignTask = TestUtils.getTask("DigitalSignatures");
+		this.generatorDigitalSIgn = new XSLBasedGenerator(testJavaProject.getProject(), digitalSignTask.getXslFile());
+		
+		this.developerProject = generatorEnc.getDeveloperProject();
+	}
+
+	/**
+	 * Test if the code generation for all CogniCrypt tasks works, without any open
+	 * class.
+	 */
+	@Test
+	public void EncDefault() {
+		this.configEnc = TestUtils.createConfigurationForCodeGeneration(developerProject, encTask);
+		boolean encCheck = generatorEnc.generateCodeTemplates(configEnc, encTask.getAdditionalResources());
+		assertTrue(encCheck);
+	}
+	
+	@Test
+	public void SecPasswordDefault() {
+		this.configSecPassword = TestUtils.createConfigurationForCodeGeneration(developerProject, secPasswordTask);
+		boolean secPasswordCheck = generatorSecPassword.generateCodeTemplates(configSecPassword, secPasswordTask.getAdditionalResources());
+		assertTrue(secPasswordCheck);
+	}
+	
+	/**
+	 * This test case is commented because it requires UI interaction
+	 */
+//	@Test
+//	public void LTADefault() {
+//		this.configLTA = TestUtils.createConfigurationForCodeGeneration(developerProject, LTATask);
+//		boolean ltaCheck = generatorLTA.generateCodeTemplates(configLTA, LTATask.getAdditionalResources());
+//		assertTrue(ltaCheck);
```

Comment:
What kind of UI interaction is required?

Reply:
@kruegers  for example, when selecting the task "Securely Archive Files", when choosing "Next>" on the Wizard, there is a part where a dialog box occurs which requires for user interaction. I don't know how to overpass the dialog box in this CogniCrypt task through the use of Plug-in tests. I am sending you also a screenshot of that dialog box.

![user_interaction_required](https://user-images.githubusercontent.com/9791632/47084067-c881e380-d212-11e8-8d73-84a16032b3e4.png)